### PR TITLE
Add Import/Export

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: slave.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+const (
+	// ErrEmptyImporter is reutrned when the importer is nil when calling NewSlave
+	ErrEmptyImporter = errors.Error("importer cannot be nil")
+)
+
+// NewSlave will return a new slave
+// Note: importInterval is interval time in seconds
+func NewSlave(name, path string, fm FuncsMap, imp Importer, importInterval int) (sp *Slave, err error) {
+	var s Slave
+	if imp == nil {
```
Comment (shahzilabid) [inline slave.go:21]: Really like the idea of a generic importer. Great stuff

File: importers/http.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	"github.com/missionMeteora/toolkit/errors"
+)
+
+// ErrEmptyLoc is returned when an empty location is provided when calling a new http importer
+const ErrEmptyLoc = errors.Error("empty importer location")
+
+// NewHTTPImporter will return a new http importer
+// Note: This is intended to be used for Slave DB's
+func NewHTTPImporter(loc string) (hp *HTTPImporter, err error) {
```
Comment (shahzilabid) [inline importers/http.go]: Hmm I can't see where you define this HTTPImporter struct?

File: importers/http.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// HTTP is an http importer to be used for Slave db's
+type HTTP struct {
+	hc  http.Client
+	url *url.URL
+}
+
+// SetJar will set a cookie jar for an HTTP importer
+func (h *HTTP) SetJar(jar http.CookieJar) {
```
Comment (OneOfOne) [inline importers/http.go]: Why use cookies instead of an api key we can pass as a header or query param?

File: importers/http.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+// Import will import from a given txnID and return a reader
+func (h *HTTP) Import(txnID string) (r io.Reader, err error) {
+	var resp *http.Response
+	if resp, err = h.hc.Get(h.url.String()); err != nil {
+		return
+	}
+
+	var buf bytes.Buffer
+	io.Copy(buf, resp.Body)
```
Comment (OneOfOne) [inline importers/http.go]: &buf, i don't think this compiles right now

**Review by OneOfOne — CHANGES_REQUESTED** [review]
great stuff man, just Shahzil's notes + 2 little notes

File: importers/http.go (hunk 3 above)
Comment (itsmontoya) [reply importers/http.go]: Cookies seems very universal for all systems. I can totally add a header option though too!

**Comment by shahzilabid** [comment]
go build -->
../../itsmontoya/mrT/mrT.go:233: m.mw.List undefined (type *middleware.MWs has no field or method List)

I'm on branch add_for_each_raw for mrT

**Comment by itsmontoya** [comment]
@shahzilabid can you try git pull on itsmontoya/middleware?

**Comment by shahzilabid** [comment]
Worked. New error: types/bytes/turtle.go:120: not enough arguments in call to t.mrT.ForEach

**Comment by itsmontoya** [comment]
[TurtleDB copyTo feature](https://trello.com/c/ixFl50z4/205-turtledb-copyto-feature)

**Comment by itsmontoya** [comment]
[TurtleDB copyTo feature](https://trello.com/c/ixFl50z4/205-turtledb-copyto-feature)

**Review by shahzilabid — APPROVED** [review]
Great job man! Really clean. Will get started on the client changes today. Will come back to you with any questions

**Review by OneOfOne — APPROVED** [review]

**Pull request merged**